libs/top_level_metrics.py

```diff
@@ -64,15 +64,23 @@ def calculate_metrics_for_timeseries(
     cumulative_cases = data[CommonFields.CASES]
     case_density = calculate_case_density(cumulative_cases, population)
 
-    cumulative_positive_tests = series_utils.interpolate_stalled_and_missing_values(
-        data[CommonFields.POSITIVE_TESTS]
-    )
-    cumulative_negative_tests = series_utils.interpolate_stalled_and_missing_values(
-        data[CommonFields.NEGATIVE_TESTS]
-    )
-    test_positivity = calculate_test_positivity(
-        cumulative_positive_tests, cumulative_negative_tests
-    )
+    if (
+        CommonFields.TEST_POSITIVITY in data.columns
+        and data[CommonFields.TEST_POSITIVITY].notna().any()
+    ):
+        # Only use TEST_POSITIVITY column if there is at least one real value in the timeseries for
+        # this region.
+        test_positivity = data[CommonFields.TEST_POSITIVITY]
+    else:
+        cumulative_positive_tests = series_utils.interpolate_stalled_and_missing_values(
+            data[CommonFields.POSITIVE_TESTS]
+        )
+        cumulative_negative_tests = series_utils.interpolate_stalled_and_missing_values(
+            data[CommonFields.NEGATIVE_TESTS]
+        )
+        test_positivity = calculate_test_positivity(
+            cumulative_positive_tests, cumulative_negative_tests
+        )
```

Comment:
I'd love to capture the test_positivity method used in the API output schema so we know how the test positivity was calculated.  Thoughts on adding that to the metrics pydantic object and passing it through similar to the icu_headroom_details?

Reply:
I have a PR also ready to go that adds the test method as a provenance value in OneRegionDataset.